src/app/item-popup/ItemSocketsGeneral.tsx

```diff
@@ -39,24 +39,47 @@ export default function ItemSocketsGeneral({
     (c) => c.category.hash === SocketCategoryHashes.Emotes,
   );
 
+  // exotic class armor intrinsics
+  const extraIntrinsicSockets =
+    item.isExotic && item.bucket.hash === BucketHashes.ClassArmor && item.sockets
+      ? item.sockets.allSockets
+          .filter((s) => s.isPerk && s.visibleInGame && socketContainsIntrinsicPlug(s))
+          .map((s) => Object.assign({}, s, { isReusable: false }))
+      : [];
+  const extraIntrinsicSocketIndices = extraIntrinsicSockets.map((s) => s.socketIndex);
+
   // Only show the first of each style of category when minimal
-  const modSocketCategories = minimal
-    ? uniqBy(modSocketsByCategory.entries(), ([category]) => category.category.categoryStyle)
-    : // This might not be necessary with iterator-helpers
-      [...modSocketsByCategory.entries()];
+  const modSocketCategories = (
+    minimal
+      ? uniqBy(modSocketsByCategory.entries(), ([category]) => category.category.categoryStyle)
+      : // This might not be necessary with iterator-helpers
+        [...modSocketsByCategory.entries()]
+  )
+    .map(
+      ([category, sockets]) =>
+        [category, sockets.filter((s) => !extraIntrinsicSocketIndices.includes(s.socketIndex))] as [
+          DimSocketCategory,
+          DimSocket[],
```

Comment:
Weird that you had to provide these types. Maybe try `as const` instead?

Reply:
Yep, that works.